test/apps/iodemo/io_demo.cc

```diff
@@ -1343,10 +1382,65 @@ class DemoClient : public P2pDemoCommon {
         }
     }
 
+    void active_servers_swap(size_t index1, size_t index2) {
+        size_t& active_server1 = _active_servers[index1];
+        size_t& active_server2 = _active_servers[index2];
+
+        std::swap(_server_info[active_server1].active_index,
+                  _server_info[active_server2].active_index);
+        std::swap(active_server1, active_server2);
+    }
+
+    void active_servers_insert(size_t server_index) {
+        assert(_last_active_server_to_use <= (_active_servers.size() - 1));
+
+        _active_servers.push_back(server_index);
+        _last_active_server_to_use++;
+
+        size_t last_index = _active_servers.size() - 1;
+        _server_info[last_index].active_index = last_index;
+
+        active_servers_swap(last_index, _last_active_server_to_use);
+        assert(_last_active_server_to_use <= last_index);
```

Comment:
can we reuse active_servers_make_used()?
e.g insert it at the end, and then active_servers_make_unused(last_index)

Reply:
done